azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -230,6 +237,11 @@ public void run() throws Exception {
             String.format("Not permitted to proxy as '%s' through Azkaban", effectiveUser)
         );
       }
+      // Set parent directory permissions to <uid>:azkaban so user can write in their execution directory
+      // if the directory is not permissioned correctly already (should happen once per execution)
+      if (!canWriteInCurrentWorkingDirectory(effectiveUser)) {
```

Comment:
This seems tricky. If the effective user is able to write to the directory, then we don't set up permissions? I think we do run into cases today where the effective user is able to write but we are unable to clean up. no?